tests/testAutopas/tests/iterators/RegionParticleIteratorTest.cpp

```diff
@@ -107,46 +116,73 @@ INSTANTIATE_TEST_SUITE_P(Generated, RegionParticleIteratorTest,
+ * Tests that AutoPas rejects regions where regionMin > regionMax.
  */
-TEST_F(RegionParticleIteratorTest, testForceSequential) {
-  constexpr size_t particlesPerCell = 1;
-  auto cells = IteratorTestHelper::generateCellsWithPattern(10, {1ul, 2ul, 4ul, 7ul, 8ul, 9ul}, particlesPerCell);
-
-  // min (inclusive) and max (exclusive) along the line of particles
-  size_t interestMin = 2;
-  size_t interestMax = 8;
-  const auto interestMinD = static_cast<double>(interestMin);
-  const auto interestMaxD = static_cast<double>(interestMax);
-  std::array<double, 3> searchBoxMin{interestMinD, interestMinD, interestMinD};
-  std::array<double, 3> searchBoxMax{interestMaxD, interestMaxD, interestMaxD};
-  std::vector<size_t> searchBoxCellIndices(interestMax - interestMin);
-  std::iota(searchBoxCellIndices.begin(), searchBoxCellIndices.end(), interestMin);
-
-  // IDs of particles in cells 2, 4, 7
-  std::vector<size_t> expectedIndices = {1, 2, 3};
-
-  constexpr size_t numAdditionalVectors = 3;
-  std::vector<std::vector<Molecule>> additionalVectors(numAdditionalVectors);
-
-  size_t particleId = cells.size() + 100;
-  for (auto &vector : additionalVectors) {
-    vector.emplace_back(Molecule({interestMinD, interestMinD, interestMinD}, {0., 0., 0.}, particleId));
-    expectedIndices.push_back(particleId);
-    ++particleId;
-  }
+TEST_F(RegionParticleIteratorTest, testInvalidBox) {
+  // setup
+  autopas::AutoPas<Molecule> autoPas{};
+  const auto [haloBoxMin, haloBoxMax] = defaultInit(autoPas, autopas::ContainerOption::directSum, 1.);
 
-#pragma omp parallel
-  {
-    std::vector<size_t> foundParticles;
-    constexpr bool modifyable = true;
-    autopas::internal::RegionParticleIterator<Molecule, FMCell, modifyable> iter(
-        &cells, searchBoxMin, searchBoxMax, searchBoxCellIndices, nullptr,
-        autopas::IteratorBehavior::ownedOrHalo | autopas::IteratorBehavior::forceSequential, &additionalVectors);
-    for (; iter.isValid(); ++iter) {
-      foundParticles.push_back(iter->getID());
-    }
-    EXPECT_THAT(foundParticles, ::testing::UnorderedElementsAreArray(expectedIndices));
-  }
+  // helpers
+  using autopas::utils::ArrayMath::mulScalar;
+  using autopas::utils::ArrayMath::sub;
+
+  // calculate box size
+  const std::array<double, 3> haloBoxLength = sub(haloBoxMax, haloBoxMin);
+  const std::array<double, 3> haloBoxLength3rd = mulScalar(haloBoxMax, 1. / 3.);
+
+  // calculate points within the domain
+  const std::array<double, 3> regionUpperLimit = mulScalar(haloBoxLength3rd, 2.);
+  const std::array<double, 3> regionLowerLimit = mulScalar(haloBoxLength3rd, 1.);
+
+  // actual test
+  EXPECT_NO_THROW(autoPas.getRegionIterator(regionLowerLimit, regionUpperLimit));
+  EXPECT_THROW(autoPas.getRegionIterator(regionUpperLimit, regionLowerLimit),
+               autopas::utils::ExceptionHandler::AutoPasException);
 }
+
+///**
+// TODO: Is this worthwhile to reimplement? If yes how?
```

Comment:
I would suggest yes. My understanding of ForceSequential, is that you may use an interator inside a parallel region. If you didn't use it, the iterator would use the number of threads as the stride, despite only one thread iterating - so some particles will be missed.

If so, I would suggest

- Create an AutoPas object, with some container configuration (auto-generated like `RegionParticleIteratorTest::testRegionAroundCorner`)
- Fill it with some particles
- Define some spatial region to iterate over, and store every ID for every particle in that region in some 'expected' vector
- Start a parallel region, with at least two threads (I think only two is suitable)
- Region iterate within parallel region with ForceSequential, check that for both threads they find all expected particles.

One potentially easy way to implement this, is to literally just use `RegionParticleIteratorTest::testRegionAroundCorner` with extra iterator options, with the expected IDs as without ForceSequential, but when ForceSequential is being tested, there is a parallel region around the call to `IteratorTestHelper::provideRegionIterator` and that a sequential version of `IteratorTestHelper::findParticles` is used.